core/app/models/spree/product.rb

```diff
@@ -36,6 +36,9 @@ class Product < Spree::Base
     has_many :product_promotion_rules, class_name: 'Spree::ProductPromotionRule'
     has_many :promotion_rules, through: :product_promotion_rules, class_name: 'Spree::PromotionRule'
 
+    has_many :promotions, through: :promotion_rules, class_name: 'Spree::Promotion'
+    has_many :advertised_promotions, ->{ Spree::Promotion.advertised }, through: :promotion_rules, class_name: 'Spree::Promotion', source: :promotion
```

Comment:
@Bharat311 - I guess we can directly use advertised scope, we don't need to write Spree::Promotion.